receiver/githubreceiver/config.go

```diff
@@ -24,19 +27,62 @@ type Config struct {
 	scraperhelper.ControllerConfig `mapstructure:",squash"`
 	Scrapers                       map[string]internal.Config `mapstructure:"scrapers"`
 	metadata.MetricsBuilderConfig  `mapstructure:",squash"`
+	WebHook                        WebHook `mapstructure:"webhook"`
+}
+
+type WebHook struct {
+	confighttp.ServerConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	Path                    string                   `mapstructure:"path"`            // path for data collection. Default is /events
+	HealthPath              string                   `mapstructure:"health_path"`     // path for health check api. Default is /health_check
+	RequiredHeader          RequiredHeader           `mapstructure:"required_header"` // optional setting to set a required header for all requests to have
+	Secret                  string                   `mapstructure:"secret"`          // secret for webhook
+}
```

Comment:
Is there any added documentation on how a user is expected to use this or how to integrate with their workflow?

Reply:
I purposefully left off documentation since this will only serve a webhook endpoint for a health check. I'll add documentation on the next PR which will begin to add attributes and minimal functionality.

Reply:
Realistically this should be done with the PR because the concern is that the docs may not actually follow the next release, but perhaps N number releases after and it leaves users figuring it out for themselves.

Reply:
yea, I didn't want any users figuring it out yet. I've got a bunch of contributions I need to work through that have been pending while I've been working on SemConv and spec stuff. But it's a super fair point, and agree. I've just pushed some docs. Feel free to give them a review. Appreciate the additional perspective here!